### ISSUE TITLE
webrtcstreamer: video from file not displaying in node client app -- WebRTC timestamp issues?

### ISSUE THREAD

**Issue by DanAndersen** [sourcey/libsourcey#128]
Hi,
I am attempting to get the webrtcstreamer sample running correctly. I am able to run the node server (Symple server listening on port 4500, Web server listening on port 4499), but when I attempt to run webrtcstreamer, my browser only displays a green screen. I am able to hear audio from my test MP4 file, but no video.

At first, I thought I had isolated the issue to a problem with timestamps in the video frames and found a workaround, but it didn't seem to fix it. Nonetheless, I'll describe the issue and how I dealt with it in case it is useful:

For each frame, I would see output in the webrtcstreamer logs:

```
[001:533] [11604] (vie_encoder.cc:511): Same/old NTP timestamp (3697567784233 <= 3698514357392) for incoming frame. Dropping.
[001:549] [11604] (vie_encoder.cc:511): Same/old NTP timestamp (3697567784266 <= 3698514357392) for incoming frame. Dropping.
...
```

It looked like WebRTC was incorrectly setting the timestamp of video frames to some point about 10 days in the past. Digging a little into WebRTC's `vie_encoder.cc` I noticed that it would manually set the timestamp when it had not been set on the initial `webrtc::VideoFrame` instance. I was able to make the "Same/old NTP timestamp" errors disappear by going into `videopacketsource.cpp` and replacing 

```
OnFrame(webrtc::VideoFrame(
            buffer, rotation_,
            packet.time), // next_timestamp_ / rtc::kNumNanosecsPerMicrosec),
            packet.width, packet.height);
```
with 

```
auto f = webrtc::VideoFrame(
			buffer, rotation_,
			packet.time);
		f.set_ntp_time_ms(std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count());
		OnFrame(f, packet.width, packet.height);
```

However, in the end this did not resolve the issue I'm having, where I'm not able to receive video output from my test video (only audio). I have tried several MP4 files on my side and all have had the same result. 

Here is a printout of the webrtcstreamer output: https://gist.github.com/DanAndersen/167f55a788ab130c2f828958e79624c6 

What I notice is that there are repeatedly several outputs like:
```
[002:511] [6972] (port.cc:1179): Jingle:Conn[000002D03494C4F0:audio:VfeG1fqk:1:0:local:udp:128.10.9.56:52178->dnOeQ3pH:1:2113937151:local:udp:128.10.9.56:52183|CRWS|0|0|9079290933588803071|1]: UpdateState(), ms since last received response=755, ms since last received data=6, rtt=100, pings_since_last_response=
[002:511] [6972] (port.cc:1179): Jingle:Conn[000002D034964FE0:audio:tIXal4z8:1:0:local:udp:192.168.56.1:52177->rHbUVDPA:1:1845501695:prflx:udp:192.168.56.1:52183|CRWS|0|0|7926369428982087167|0]: UpdateState(), ms since last received response=250, ms since last received data=2204, rtt=100, pings_since_last_response=
[002:512] [6972] (port.cc:1179): Jingle:Conn[000002D034968640:audio:tIXal4z8:1:0:local:udp:192.168.56.1:52177->dnOeQ3pH:1:2113937151:local:udp:128.10.9.56:52183|C--I|0|0|9079290933588934143|-]: UpdateState(), ms since last received response=950193105, ms since last received data=950193105, rtt=6000, pings_since_last_response=50716c47466b686351615132 514141664c51536558553568 43732f426a466a6a49785961 666f4f70726e6d4844785433
```

That "ms since last received response/data" having a value around "950193105" is suspicious, since it's about the same amount of time in between those incorrect NTP timestamps I kept on getting.

In short, I suspect there's something wrong in how WebRTC is figuring out timestamps, but I don't know if it's something we can fix in the libsourcey codebase. Have you encountered anything like this?

**Comment by auscaster** [comment]
Hi Daniel

I've just pushed a big update to the webrtcstreamer sample to bring it up to date with the current webrtc API. It's working a treat on my end, hope it sorts your issues out too.

**Comment by DanAndersen** [comment]
Thanks! I've pulled down the latest commit and (after a quick modification) have been able to get the WebRTC streaming example to work flawlessly. 

(I was getting compile errors in `cipher.h` and found it was because the latest commit had removed `#include "scy/uv/uvpp.h" // ssize_t`)

**Comment by auscaster** [comment]
Nice one ... I'll make sure scy/uv/uvpp.h is added back in also (just
needed on windows apparently)

**Comment by auscaster** [comment]
Hey good on you for sticking with it ... are you going to build something
cool now? :D

**Comment by DanAndersen** [comment]
Yep! As part of a research project I'm having a C++ app that will receive/display video coming from a remote head-mounted camera. For that reason I'm currently looking through the webrtcrecorder sample to try to figure out how to receive frames and decode them as OpenCV Mats.

**Comment by auscaster** [comment]
OK cool ... yeah the webrtcrecorder should have most of what you need. You can modify StreamRecorder to put the raw data into a cv::Mat instead of send it to FFmpeg and you're pretty much there!

**Comment by auscaster** [comment]
Some updates here you might be interested in: https://github.com/sourcey/libsourcey/commit/fdd498d44f3d46ba1ab07df6ce1250fec0ed5ede
